lib/auth/auth.go

```diff
@@ -5371,6 +5380,79 @@ func (a *Server) syncDynamicLabelsAlert(ctx context.Context) {
 	}
 }
 
+func (a *Server) cleanupNotifications(ctx context.Context) {
+	userNotifications, _, err := a.Cache.ListUserNotifications(ctx, 0, "")
+	if err != nil {
+		slog.WarnContext(ctx, "failed to list first set of notifications for periodic cleanup", err)
+	}
+	globalNotifications, _, err := a.Cache.ListGlobalNotifications(ctx, 0, "")
+	if err != nil {
+		slog.WarnContext(ctx, "failed to list second set of notifications for periodic cleanup", err)
+	}
+
+	timeNow := time.Now()
+
+	// Convert into a map where the key is the notification id.
+	globalNotificationsIdMap := make(map[string]*notificationsv1.GlobalNotification)
```

Comment:
```suggestion
	globalNotificationsIDMap := make(map[string]*notificationsv1.GlobalNotification)
```

Or even `globalNotificationsByID` (we don't really need to put `map` in the variable name)